analyses/hotspots-detection/utils/prepMaf.R

```diff
@@ -0,0 +1,113 @@
+#' Takes a sqllite table from an individual caller in maf format
+#' and filters as per given user filtering options
+#' @param table prefered sqllite table with individual caller maf or a maf format dataframe
+#' @param variant_classification_values [optional] list of Variant_Classification values to filter
+#' @param impact_values list of IMPACT [optional] values to filter
+#' @param gene_table [optional] gene df with columns "Hugo_Symbol" and "type"
+#' @param hotspot_amino_acid_position_df  [optional] dataframe with Hugo_Sybol, Amino_Acid_Position 
+#' and hotspot_database columns
+#' @param hotspot_genomic_site_df [optional] dataframe with Hugo_Symbol and Chromosome, Start_Position , 
+#' End_Position, Hugo_Symbol and hotspot_database columns 
+#' @return A dataframe with base columns from maf filtered as per user input for Variant_Classification 
+#' column with "variant_classification" OR IMPACT column with "impact_values" OR Hugo_Symbol using "gene_table".
+#' Additional filtering can be done per amino acid position hotspots using "hotspot_amino_acid_site_df" 
+#' OR per genomic site hotspots using "hotspot_genomic_site_df"
+#'
+
+prepMaf <- function(table,
+                    variant_classification_values,
+                    impact_values,
+                    gene_table,
+                    hotspot_amino_acid_position_df, 
+                    hotspot_genomic_site_df  ) {
+  if (!missing(gene_table)){
+  # gene list to filter sql lite table
+  gene_list <- gene_table$Hugo_Symbol
+  table <- table %>%
+    dplyr::filter(Hugo_Symbol %in% gene_list)
+  }
+
+  
+  
+  if (!missing(variant_classification_values)){
+  # Variant_Classification filtering values
+  variant_classification_values <- unlist(str_split(variant_classification_values,pattern = "[|]"))
+  table <- table %>%
+    dplyr::filter(Variant_Classification %in% variant_classification_values)
+  }
+  
+  if (!missing(impact_values)){
+  # IMPACT filtering values
+  impact_values <- unlist(str_split(impact_values,pattern = "[|]"))
+  table <- table %>%
+    dplyr::filter(IMPACT %in% impact_values)
+  }
+  
+  # filtered calls_base 
+  calls_base <- table %>%
+    as.data.frame() %>%
```

Comment:
I thought of keeping it table/dataframe as input since I was using the same script for the sql database and directly reading the caller maf files here. I guess now that the sql database PR is closed I can update it to be dataframe input only